### PR TITLE
Add take while overload

### LINKED ISSUE #1203: Add "take while" function overload

### PR DESCRIPTION
**Pull request by Steffan153**
Closes #1203

### CODE REVIEW

File: vyxal/elements.py [hunk 1]
```diff
@@ -6115,6 +6115,19 @@ def zero_slice(lhs, rhs, ctx):
     (num, any) -> b[0:a]
     (str, str) -> regex.findall(pattern=a,string=b) (Find all matches for a regex)
     """
+    if type(lhs) == types.FunctionType:
+        return zero_slice(rhs, lhs, ctx)
+    if type(rhs) == types.FunctionType:
+        i = iterable(lhs, ctx=ctx)
```
Comment (cgccuser) [inline vyxal/elements.py:6121]: I don't wanna be that guy, but could you name this something more descriptive? Not sure what `i` stands for

**Review by cgccuser — CHANGES_REQUESTED** [review]
Huh, I thought we had this already. A drop while element would also be nice. LGTM other than one minor thing

File: vyxal/elements.py (hunk 1 above)
Comment (Steffan153) [reply vyxal/elements.py:6121]: `i` stands for `iterable` lol

Comment (cgccuser) [reply vyxal/elements.py]: Oh ok

**Review by cgccuser — APPROVED** [review]

**Review by chunkybanana — APPROVED** [review]

**Comment by chunkybanana** [comment]
this somehow made it into main already. Make sure to fork your branches from main @Steffan153

**Pull request merged**